test/libsolutil/JSON.cpp

```diff
@@ -34,58 +34,58 @@ BOOST_AUTO_TEST_SUITE(JsonTest, *boost::unit_test::label("nooptions"))
 
 BOOST_AUTO_TEST_CASE(json_types)
 {
-	auto check = [](Json::Value value, std::string const& expectation) {
+	auto check = [](Json value, std::string const& expectation) {
 		BOOST_CHECK(jsonCompactPrint(value) == expectation);
 	};
 
-	Json::Value value;
+	Json value;
 	BOOST_CHECK(value.empty());
 	value = {};
 	BOOST_CHECK(value.empty());
-	value = Json::Value();
+	value = Json();
 	BOOST_CHECK(value.empty());
-	value = Json::nullValue;
+	value = Json();
 	BOOST_CHECK(value.empty());
 
 	check(value, "null");
 	check({}, "null");
-	check(Json::Value(), "null");
-	check(Json::nullValue, "null");
-	check(Json::objectValue, "{}");
-	check(Json::arrayValue, "[]");
-	check(Json::UInt(1), "1");
-	check(Json::UInt(-1), "4294967295");
-	check(Json::UInt64(1), "1");
-	check(Json::UInt64(-1), "18446744073709551615");
-	check(Json::LargestUInt(1), "1");
-	check(Json::LargestUInt(-1), "18446744073709551615");
-	check(Json::LargestUInt(0xffffffff), "4294967295");
-	check(Json::Value("test"), "\"test\"");
+	check(Json(), "null");
+	check(Json(), "null");
+	check(Json::object(), "{}");
+	check(Json::array(), "[]");
+	check(1, "1");
+	check(static_cast<uint32_t>(-1), "4294967295");
+	check(1, "1");
+	check(static_cast<uint64_t>(-1), "18446744073709551615");
+	check(1, "1");
+	check(static_cast<uint64_t>(-1), "18446744073709551615");
+	check(0xffffffff, "4294967295");
+	check(Json("test"), "\"test\"");
 	check("test", "\"test\"");
 	check(true, "true");
 
-	value = Json::objectValue;
+	value = Json::object();
 	value["key"] = "value";
 	check(value, "{\"key\":\"value\"}");
 
-	value = Json::arrayValue;
-	value.append(1);
-	value.append(2);
+	value = Json::array();
+	value.push_back(1);
+	value.push_back(2);
 	check(value, "[1,2]");
 }
 
 BOOST_AUTO_TEST_CASE(json_pretty_print)
 {
-	Json::Value json;
-	Json::Value jsonChild;
+	Json json;
+	Json jsonChild;
 
 	jsonChild["3.1"] = "3.1";
 	jsonChild["3.2"] = 2;
 	json["1"] = 1;
 	json["2"] = "2";
 	json["3"] = jsonChild;
 	json["4"] = "ऑ ऒ ओ औ क ख";
-	json["5"] = "\xff";
```

Comment:
`\xff` is not a valid unicode sequence - so I needed to change it

Reply:
We'll need to think through consequences of this... are there any cases in which there was valid input involving something like this for which the behaviour changes? Probably fine, but worth a bit of thought.

Reply:
I just noted that here I just got a bit confused: though `\xff` is an invalid utf8 sequence, the sequence that I put here was not anything special anymore, because I just escaped the `\x` wrongly `\\xff` that made it just work by accident. 

I also thought that this `\x` is also valid in json, but json doesn't allow any `\x`, because this can only be done with unicode escapes like e.g. `\u0010`.

However, this is exactly the difference: `jsoncpp` is allow arbitrary character sequences, where `nlohmann` is checking those - they need to be valid utf8 sequences.

Reply:
Alright - but I guess this is not really an issue, right?
I mean, we do allow non-ascii and unicode inputs.
Like
```
{
        "language": "Solidity",
        "sources":
        {
                "A":
                {
                        "content": "pragma solidity >=0.0; contract C { function f() public pure returns (bytes memory) { return unicode'z'; } }"
                }
        }
}
```
I just paranoidly tested this - if I pipe that through ``sed -e "s/z/Z\xff/"``, replacing the string content with an invalid unicode sequence, I get ``SyntaxError: Contains invalid UTF-8 sequence at position 1.``.
I guess this would already hit an error sooner now with nlohmann-json and already fail at json-parsing-time?

That would be fine then, if this just moves the error, but we just turn error cases in different error cases.

Apart from that the only thing I could imagine would be things outside the sources like the contract name. If I pipe the above through ``sed -e "s/A/\xff/"``, the result unforunately compiles... it seems to interpret the source name as ``\ufffd"`` in the output...

I guess that also fails with the nlohmann branch now?

I mean, technically, strictly speaking, that'd make this breaking... but we may get around arguing that supporting non-unicode non-ascii contract names is more of a bug than a feature...
Do people these days still use non-unicode-non-ascii encodings for anything?

It's actually also curious that it becomes ``\ufffd`` with json-cpp... does json-cpp try to guess and fix invalid unicode sequences there and reinterprets them as the closest it finds? If so, we could definitely call it a bugfix...

But yeah, we should spend a bit more thought about this here and about these cases, but in the end I think we can get away with it.